lib/TPP/ConvertLinalgToTpp.cpp

```diff
@@ -288,8 +286,11 @@ struct ConvertGenericOpToTpp : public OpRewritePattern<linalg::GenericOp> {
       return success();
     }
     if (tpp::utils::isTppAdd(linalgOp)) {
+      // A = A + B
```

Comment:
IIRC from the discussion yesterday, `A = A + B` requires `A.shape == B.shape`, but for `C = A + B`, we could have `A.rank < B.rank`, with automatic broadcast (if the lower rank dims match).

The definition above uses `SameTypeOperands`, so the latter case would be invalid, which is ok as a first approximation, but if/when we have that case, we'll probably need to have an explicit broadcast before the TPP operations, and match both ops to the auto-broadcast TPP op.